src/App.tsx

```diff
@@ -14,11 +15,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[];
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  addToSelectedGood = (good: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  };
+
+  removeToSelectedGood = (good: string) => {
+    const { selectedGoods } = this.state;
+
+    if (selectedGoods.includes(good)) {
+      const index = selectedGoods.lastIndexOf(good);
+
+      selectedGoods.splice(index, 1);
+    }
+
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods],
+    }));
+  };
+
+  resetState = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const { addToSelectedGood, removeToSelectedGood, resetState } = this;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">
+          {selectedGoods.length
+            ? (`${selectedGoods.join(', ')} selected`)
+            : ('No goods selected')}
+        </h1>
+        {selectedGoods.length
+          ? (
+            <button
+              className="card__btn card__btn--reset"
+              type="button"
+              onClick={resetState}
+            >
+              Clear
+            </button>
+          )
+          : null}
+        <div className="container">
+          {goodsFromServer.map(good => (
+            <div className={classNames(
+              'card',
+              { card__active: selectedGoods.includes(good) },
+            )}
+            >
+              <h2 className="card__title" key={good}>
```

Comment:
if i am not mistaken, key attribute should go on the first child element

Reply:
fixed)